src/js/nav/buildNav.js

```diff
@@ -0,0 +1,74 @@
+const yaml = require('js-yaml');
+const axios = require('axios');
+const _ = require('lodash');
+const fs = require('fs');
+
+axios.get('https://raw.githubusercontent.com/RedHatInsights/cloud-services-config/enhancements/chrome-nav/main.yml')
+.then(response => yaml.safeLoad(response.data)).then(data => buildNavFromConfig(data));
+
+function buildNavFromConfig(masterConfig) {
+    let globalNav = {};
+    // Get the top-level apps from the master config
+    masterConfig.filter(app => app.top_level).forEach((app) => {
+        globalNav[app.id] = {
+            title: app.title
+        };
+        globalNav[app.id].routes = getRoutesForApp(app, masterConfig);
+    });
+
+    // Write to appropriate file
+    fs.writeFile('src/js/nav/globalNav.json', JSON.stringify(globalNav), 'utf8', function(err) {
+        if (err) {
+            throw err;
+        }
+    });
+
+}
+
+// Returns a list of routes/subItems owned by an app
+function getRoutesForApp(app, masterConfig) {
+    let routes = [];
+    if (_.has(app, 'sub_apps')) {
+        app.sub_apps.forEach((subApp => {
+            let subAppData = getAppData(subApp.id, 'subItems', masterConfig);
```

Comment:
What parts of `getAppData` are you actually interested in? And what is the shape of some object? It would be much better to use something different than delete.

Reply:
OK. Yeah, now that we have even more fields added to it, I'm going to have to make that more precise instead of deleting things afterwards.